src/lib/es5.d.ts

```diff
@@ -513,7 +513,7 @@ interface Boolean {
 
 interface BooleanConstructor {
     new(value?: any): Boolean;
-    <T>(value?: T): value is Exclude<T, false | null | undefined | '' | 0>;
+    <T extends any>(value?: T): value is Exclude<T, false | null | undefined | '' | 0>;

```

Comment:
https://github.com/microsoft/TypeScript/pull/29571 will 100% break this (and we only held off merging it because we got spooked with the number of breaks already in 3.5 IIRC), since this is a terrible hack that makes `T` "look like any" even though it's a type parameter and `extends any` should be identical to `extends unknown` or simply no constraint.